Feat: Fast Stop

H3XORSauvage opened the pull request:
# Feature: Fast Stop

This pull request introduces a new version of the unused "Fast Stop" feature, designed to enhance player movement control and accuracy within the game.

## Description

The "Fast Stop" feature allows players to instantly halt their momentum, eliminating the sliding effect typically associated with stopping in-game. This provides a significant advantage by enabling precise and immediate stops, which can be crucial for:

*   **Improved Accuracy:** By stopping instantly, players can achieve maximum shooting accuracy much faster, leading to more consistent shots.
*   **Faster Peeking:** Enables quick peeks and re-peeks around corners without overshooting or exposing oneself for too long.
*   **Enhanced Movement Control:** Provides a more responsive and predictable movement experience.

## Implementation Details

This feature is primarily implemented in `DragonBurn/Features/Misc.cpp` and declared in `DragonBurn/Features/Misc.h`. The `FastStop()` function is called within the main cheat loop via `MiscFuncs` in `DragonBurn/Core/Cheats.cpp`.

The mechanism involves detecting when movement keys (W, A, S, D or Z, Q, S, D for AZERTY) are released and then simulating a brief press of the opposite movement key to instantly stop the player's momentum. It includes logic to prevent activation when the Spacebar (jump) or Left Shift (walk) keys are held down, ensuring natural movement in those contexts. A configurable delay (`MiscCFG::FastStopDelay`) is also incorporated to fine-tune the stopping behavior.

## How to Use

To enable this feature, ensure `MiscCFG::FastStop` is set to `true` in the configuration. The feature automatically activates when you release movement keys. You can adjust the `MiscCFG::FastStopDelay` to control the duration of the simulated stop. The feature will not activate if you are holding down the Spacebar (jump) or Left Shift (walk) keys.
<img width="599" height="498" alt="Screenshot_15" src="https://github.com/user-attachments/assets/d528acec-bb1d-4e83-b00b-92716704bd05" />

## Testing

The feature has been tested to ensure stability and proper functionality without causing any unintended side effects or conflicts with other features.

## Checklist

*   [x] Feature works as intended.
*   [x] Automatic AZERTY / QWERTY Keyboard
*   [x] No regressions introduced.
*   [x] Configuration options are clear and functional.

ByteCorum committed:
Merge pull request #206 from ByteCorum/WIP

Minor update

ByteCorum committed:
Merge pull request #208 from ByteCorum/WIP

Hot fix

ByteCorum committed:
Merge pull request #211 from ByteCorum/WIP

Minor update

ByteCorum committed:
Merge pull request #216 from ByteCorum/WIP

Minor update

ByteCorum committed:
Update README.md

NoPaperSpray committed:
FastStop

ByteCorum commented:
What fast stop delay exactly is

ByteCorum reviewed (changes requested):
- Give pls understandable names to vars
- If fast stop is off, it shouldn't even start to execute the func, instant return
- Add UI description for fast stop delay

ByteCorum reviewed (changes requested):
I've reviewed ur code again, and it's a bit messy. Pls don't use lambda func. And azerty in config looks like stub

H3XORSauvage commented:
I made the changes but I'm totally lost when it comes to re-push. I'm looking for solutions on internet.

estdae commented:
you can push directly to the pull request

H3XORSauvage commented:
I tried but idk why it's not working ;_;

NoPaperSpray committed:
Update code PR #220

H3XORSauvage commented:
Now, i think it's fixed.

NoPaperSpray committed:
Automatic keyboard detection

ByteCorum reviewed (changes requested):
Also, pls test ur code and provide the result of tests

ByteCorum commented:
and resolve conflicts pls

ByteCorum reviewed (changes requested):
also add default values in configmenu.cpp

ByteCorum commented:
also how about removing delay, why do we need it

H3XORSauvage commented:
> also how about removing delay, why do we need it

The FastStopDelay has two main functions:

Counter-pulse duration:
It defines how long the opposite movement key is pressed to stop the player.

If the duration is too short, it may not be enough to fully cancel the momentum.

If it’s too long, it may cause unwanted movement in the opposite direction.

Cooldown time:
It prevents the function from triggering repeatedly and unintentionally when you press movement keys rapidly, which could otherwise result in jittery or choppy movement.

The delay therefore fine-tunes the "fast stop" behavior so that it remains both effective and smooth.
Removing it could make the feature less reliable

NoPaperSpray committed:
New KeyBoard Layout and optimisation, add default values in configmenu.cpp

ByteCorum reviewed (changes requested):
- In file `DragonBurn/Features/Misc.cpp\Misc.h` keyboard layout code is not full and it repeats itself
- It gonna be hard for user to select best delay, so just make it constant with best value
- And fix conflicts or if u can't, I'll do it
- Otherwise, everything is ready to be merged

H3XORSauvage commented:
i will fix the keyboard thing, and remove the delay slidder.
For conflict, im not very good with github, I'm afraid of doing something stupid.
Is it okay if I do this tomorrow?

NoPaperSpray committed:
Fixed Keyboard Layouts thing, removed slidder for fast stop delay.

H3XORSauvage commented:
I created a new issue with keyboard layout that is not being recognised
I will fix that

NoPaperSpray committed:
The keyboard layout is now detected automatically at each startup and is no longer saved in or
     loaded from configuration files. This prevents a saved configuration from overriding the correct,
     automatically detected layout.

NoPaperSpray committed:
Merge branch 'WIP' into new_branch

ByteCorum commented:
is it ready for review?

ByteCorum reviewed (approved):
I'll merge it soon

H3XORSauvage commented:
Yes it is

ByteCorum committed:
refactor: useless redefinitions of `KeyboardLayout`

GPT code is such a shit, won't be approved any more
